src/main/java/com/zaxxer/hikari/pool/PoolBase.java

```diff
@@ -332,17 +332,17 @@ Connection newConnection() throws Exception
     */
    private void setupConnection(final Connection connection) throws SQLException
    {
+      checkDriverSupport(connection);
```

Comment:
Not sure if this should be executed before or after timeout is set, I guess now it might allow longer execution in checkDriverSupport SQL. Maybe move this down again?